motoman_gp25_support/urdf/gp25_macro.xacro

```diff
@@ -0,0 +1,165 @@
+            </geometry>
+            <material name="yaskawa_blue"/>
+        </visual>
+        <collision>
+            <geometry>
+                <mesh filename="package://motoman_gp25_support/meshes/collision/gp25_link_5_b.stl"/>
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}link_6_t">
+        <visual>
+            <geometry>
+                <mesh filename="package://motoman_gp25_support/meshes/visual/gp25_link_6_t.stl"/>
+            </geometry>
+            <material name="yaskawa_blue"/>
+        </visual>
+        <collision>
+            <geometry>
+                <mesh filename="package://motoman_gp25_support/meshes/collision/gp25_link_6_t.stl"/>
+            </geometry>
+        </collision>
+    </link>
+    <link name="${prefix}tool0"/>
+    <!-- end of link list -->
+
+    <!-- joint list -->
+    <joint name="${prefix}joint_1_s" type="revolute">
+        <parent link="${prefix}base_link"/>
+        <child link="${prefix}link_1_s"/>
+        <origin xyz="0 0 0.505" rpy="0 0 0" />
+        <axis xyz="0 0 1" />
+        <limit lower="-3.1416" upper="3.1416" effort="3087" velocity="3.6651"/>
+    </joint>
+    <joint name="${prefix}joint_2_l" type="revolute">
+        <parent link="${prefix}link_1_s"/>
+        <child link="${prefix}link_2_l"/>
+        <origin xyz="0.150 0 0" rpy="0 0 0" />
+        <axis xyz="0 1 0" />
+        <limit lower="-1.8326" upper="2.7052" effort="3087" velocity="3.6651"/>
+    </joint>
+    <joint name="${prefix}joint_3_u" type="revolute">
+        <parent link="${prefix}link_2_l"/>
+        <child link="${prefix}link_3_u"/>
+        <origin xyz="0 0 0.760" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="-1.5009" upper="2.7925" effort="926.1" velocity="4.6251"/>
+    </joint>
+    <joint name="${prefix}joint_4_r" type="revolute">
+        <parent link="${prefix}link_3_u"/>
+        <child link="${prefix}link_4_r"/>
+        <origin xyz="0.795 0 0.200" rpy="0 0 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="-3.4906" upper="3.4906" effort="88.2" velocity="7.3303"/>
+    </joint>
+    <joint name="${prefix}joint_5_b" type="revolute">
+        <parent link="${prefix}link_4_r"/>
+        <child link="${prefix}link_5_b"/>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <axis xyz="0 -1 0" />
+        <limit lower="-2.6179" upper="2.6179" effort="70.56" velocity="7.3303"/>
+    </joint>
+    <joint name="${prefix}joint_6_t" type="revolute">
+        <parent link="${prefix}link_5_b"/>
+        <child link="${prefix}link_6_t"/>
+        <origin xyz="0 0 0" rpy="0 0 0" />
+        <axis xyz="-1 0 0" />
+        <limit lower="-7.9412" upper="7.9412" effort="50.31" velocity="15.4461"/>
+    </joint>
+    <joint name="${prefix}joint_6_t-tool0" type="fixed">
+        <origin xyz="0.100 0 0" rpy="3.1415926 -1.570796 0"/>
+        <parent link="${prefix}link_6_t"/>
+        <child link="${prefix}tool0"/>
+    </joint>
```

Comment:
Typically joint T is translated in the X-axis from joint B, and the `tool0` frame only has a rotation from joint T. There is nothing technically incorrect about this approach, but you might consider changing this to be consistent with the other robot models

Reply:
As above, it matches the standard Yaskawa model construction.  It then makes it easier to create models and compare them with Yaskawa's MotoSim software.

Reply:
@EricMarcil: does the internal model of the robot really translate over both the X and Z axis between joints `R` and `B`?

Reply:
Oops, it seems that my comment above that I was referencing was never posted but it's basically the same topic.
There is a translation in both X and Z between the U and R axes (like the physical joint).  We use DH representation so it corresponds to the "a" and "d" offsets.  In our model, the R, B and T axes origins all super-impose on the same spot in space (we refer to it a the "P" point, which is in the middle of the wrist.  We then have an offset between the "T-axis" and the robot flange, which corresponds to the "Tool 0".  All the Yaskawa models that I have build recently are like that.

Reply:
Thanks for the clarification.

It would still be advisable to separate the translations and only have one per dimension, as this seems to help tools like IKFast generate IK solvers (at least, it did last time I checked).

I would recommend to do what @marip8 suggested.

Reply:
Ok, if you think it may help the IK Solvers.  I'll have to adjust the Mesh to relocate the origin of the model.  I'll try to get it done today.